src/core/common/message.cpp

```diff
@@ -801,11 +801,9 @@ Message *MessageQueue::GetHead(void) const
     return (GetTail() == NULL) ? NULL : GetTail()->Next();
 }
 
-otError MessageQueue::Enqueue(Message &aMessage, QueuePosition aPosition)
+void MessageQueue::Enqueue(Message &aMessage, QueuePosition aPosition)
 {
-    otError error = OT_ERROR_NONE;
-
-    VerifyOrExit(!aMessage.IsInAQueue(), error = OT_ERROR_ALREADY);
+    VerifyOrExit(!aMessage.IsInAQueue(), OT_NOOP);
```

Comment:
```suggestion
    VerifyOrExit(aMessage.GetMessageQueue() != this, OT_NOOP);
```

If `aMessage` belongs to another `MessageQueue`, better to just let it fail in the `OT_ASSERT` below?